polynomial.py

```diff
@@ -69,5 +74,59 @@ def get_centered_remainder(x, modulus):
     return r if r <= modulus / 2 else r - modulus
 
 
+def poly_div(dividend: list[int], divisor: list[int]):
+    dividend = [int(x) for x in dividend]
+
+    # Initialize quotient and remainder
+    quotient = [0] * (len(dividend) - len(divisor) + 1)
+    remainder = list(dividend)
+
+    # Main division loop
+    for i in range(len(quotient)):
+        coeff = (
+            remainder[i] // divisor[0]
+        )  # Calculate the leading coefficient of quotient
+        # turn coeff into an integer
+        coeff = coeff
+        quotient[i] = coeff
+
+        # Subtract the current divisor*coeff from the remainder
+        for j in range(len(divisor)):
+            rem = remainder[i + j]
+            rem -= divisor[j] * coeff
+            remainder[i + j] = rem
+
+    # Remove leading zeroes in remainder, if any
+    while remainder and remainder[0] == 0:
+        remainder.pop(0)
+
+    return quotient, remainder
+
+
+def poly_mul(poly1: list[int], poly2: list[int]):
```

Comment:
it's a better practice to check `poly1` and `poly2`'s leading coefficients are non-zero (assuming that 0th element = highest degree coefficient)